flake.nix

```diff
@@ -0,0 +1,86 @@
+{
+  description = "Settings application for the COSMIC desktop environment";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    nix-filter.url = "github:numtide/nix-filter";
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, nix-filter, crane, fenix }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" "aarch64-linux" ] (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        craneLib = crane.lib.${system}.overrideToolchain fenix.packages.${system}.stable.toolchain;
+        crateNameFromCargoToml = craneLib.crateNameFromCargoToml {cargoToml = ./app/Cargo.toml;};
+        pkgDef = {
+          inherit (crateNameFromCargoToml) pname version;
+          src = nix-filter.lib.filter {
+            root = ./.;
+            include = [
+              ./app
+              ./page
+              ./pages
+              ./Cargo.toml
+              ./Cargo.lock
+              ./i18n
+              ./i18n.toml
+              ./justfile
+              ./resources
+            ];
+          };
+          nativeBuildInputs = with pkgs; [
+            just
+            pkg-config
+            autoPatchelfHook
+          ];
+          buildInputs = with pkgs; [
+            libxkbcommon
+            glib
+            gtk4
```

Comment:
Is glib and gtk4 necessary? We only need these for debian packaging:

- cmake
- libexpat1-dev
- libfontconfig-dev
- libfreetype-dev lld
- pkg-config
- libxkbcommon-dev

Reply:
Ahh ya, I mostly just copied one of our exisitng flakes for this. I'll see if i can clean it up a bit.